Week1/MAKEME.md

```diff
@@ -35,33 +35,106 @@ Write commands to do following:
+  // TODO -> assign a new value to your variable x
+  console.log('the value of my string will be: whateverYouThinkItWillLog')
+  // TODO -> log the actual value of the string to the console
+  ```
+
+3. How do you round the number 7.25, to the nearest integer?
+  3.1 Declare a variable `z` and assign the number 7.25 to it.
+  3.2 Console.log `z`.
+  3.3 Now round `z` to the nearest integer.
+  3.4 Console.log `z` again.
+
+4. Arrays!
+  4.1 Declare an empty array (you can decide on how to call it yourself).
+  4.2 Write a console.log statement in which explain what you think the value of the array is.
+  4.3 Console.log your array.
+  4.4 Create an array that has your favorite animals inside
+  4.5 Log your array
+  4.6 Ads Daan's favorite animal (baby pig) to the existing array
+  4.7 Log your new array!
+  4.8 Now add Gijs's favorite animal to the array (moose), but make sure it will be placed after the first animal that you have placed in your array.
+  4.9 Write a console.log statement in which explain what _you think_ the new value of the array is.
+  4.10 Log your new new array!
+  4.11 Log the length of the array, add a message: "The array has a length of: "(here you should show the length of the array)
+  4.12 Eyad does not like giraffes, delete this animal from the array
+  4.13 Again log your new array.
+  4.14 Now if unlike Gijs, you don't like mooses and you want to delete it from the array, but you don't know the position or the `index` of the item in the array, how can you find it?
+  4.15 Log the index of moose to the console. Add a message so it says: "The item you are looking for is at index: " (here you should show the index of the item)
+
+5. More strings 
+  5.1 Let's consider the following string: `let mystring = "this,is,a,test"`.
+  5.2 Add the string to your file and console.log it.
+  5.3 Find a way to get the length of `mystring`.
+  5.4 Console.log the length of `mystring`.
+  5.5 The comma's make that the sentence is quite hard to read. Find a way to remove the comma's from the sting and replace them with a spaces
+  5.6 Console.log `mystring` to see if you succeeded. 
+
+
+7. Write a program that checks the types of two variables and prints out `SAME TYPE` if they are the same type. 
+  6.1 First declare at least four variables and assign them with different data types.
+  6.2 For each variable write a console log statement that that logs the value 
+  ```js 
+  let foo = 3
+  console.log('My the value of my variable foo is: ' + foo)
+  ```
+  6.3 Now write a console.log statement weiring you first explain what you think the _type_ of your variables is.
+  6.4 Now use `typeof` to log the actual _type_ of your variables.
+  6.5 Now compare the types of your different variables with one another.
+  6.6 Make sure to also show a message when the variables you are comparing are not the same type.
+  For example:    
     ```js
-    let x = 9;
-    let y = 'Hello';
+    let x = 9
```

Comment:
BTW are we doing semicolons or not? I remember from last discussion (don't remind Mauro) that we decided to use semi-colons, although I don't use them myself. Strongest supporter for semicolons was Rob but he out of the picture for the time being. The book that is on the reading list (Speaking JavaScript) actually recommends semicolons.

Reply:
I've added semicolons :)